src/backend/distributed/commands/statistics.c

```diff
@@ -445,19 +446,30 @@ GetExplicitStatisticsCommandList(Oid relationId)
 
 		if (alterStatisticsOwnerCommand != NULL)
 		{
-			alterStatisticsCommandList =
-				lappend(alterStatisticsCommandList,
+			explicitStatisticsCommandList =
+				lappend(explicitStatisticsCommandList,
 						makeTableDDLCommandString(alterStatisticsOwnerCommand));
 		}
 	}
 
+	/* find indexes on current relation, in case of modified stats target */
+	Relation relation = relation_open(relationId, AccessShareLock);
+	List *indexOidList = RelationGetIndexList(relation);
+	relation_close(relation, AccessShareLock);
```

Comment:
better not to release lock here, as we still rely on relation index'es, `NoLock` should work